.github/workflows/test-pythonpath-ubuntu.yml

```diff
@@ -0,0 +1,133 @@
+      matrix:
+        # Python 3.12 cannot be tested until Ray provides support:
+        #   https://github.com/ray-project/ray/issues/40211
+        # This tests some very basic and long-lived Python behavior, so it is
+        # likely sufficient to test only on a single Python version to avoid
+        # GitHub worker contention.
+        python-version: ["3.11"]
+        pytest-flags: ["-m unmarked", "-m slow", "-m examples", "-m hypertrace"]
+
+    steps:
+
+      - uses: actions/checkout@v3
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v3
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+
+      - name: Install ISOFIT and most software dependencies
+        shell: bash
+        run: |
+
+          # Python packaging dependencies. The presence of 'wheel' triggers
+          # a lot of new Python packaging machinery that will eventually
+          # become the default.
+          python3 -m pip install pip setuptools wheel --upgrade
+          python3 --version
+          python3 -m pip show pip setuptools wheel
+
+          # Install 'isofit' and dependencies. Be sure to uninstall this version
+          # of 'isofit' to ensure we are testing against '$PYTHONPATH'.
+          python3 -m pip install ".[test]"
+          
+          # Uninstall 'isofit', leaving its dependencies.
+          python3 -m pip uninstall --yes isofit
+          
+          # Move 'isofit' into a new directory to ensure it is not just
+          # inherently available.
+          mkdir isolation/
+          mv isofit/ isolation/
+          
+          # Check to see if 'isofit' is still installed. It should not be.
+          # Note that this job is implicitly configured with 'set -e', which
+          # means that any command exiting with a non-zero exit code will cause
+          # the script to halt. The easiest way to see if 'isofit' is still
+          # available is to attempt to import it, however, that rightfully
+          # causes Python to exit with a non-zero exit code. It is possible to
+          # use a heredoc to wrap this import in a 'try/except', print a status,
+          # and have Python exit with a 0 exit code, but all of the options are
+          # painful.
+          #
+          # Instead, disable 'set -e' by calling 'set +e', but be sure to
+          # re-enable by calling 'set -e' again as soon as possible!!
+          set +e
+          ISOFIT_INSTALLED_PATH=$(
+            python3 -c "import isofit; print(isofit.__file__)" 2> /dev/null)
+          ISOFIT_INSTALLED_EXIT_CODE=$?
+          set -e
+          if [ "${ISOFIT_INSTALLED_EXIT_CODE}" -ne 1 ]; then
+              exit 1
+          fi
+          
+          # Install 'isofit' via '$PYTHONPATH', and ensure it can be imported.
+          export PYTHONPATH="isolation/:${PYTHONPATH}"
+          python3 -c "import isofit; print(isofit.__file__)"
+
+      - name: Cache 6S
```

Comment:
I believe there is a way to share parts of GitHub actions workflows between yaml files, but it was not immediately obvious to me how to do it. IMO getting good test coverage quickly is more important than eliminating duplicate code.

Reply:
Yeah, I looked at that. If you're not aware, the `action/name_of_action@vx` syntax is actually `organization/repository@tag`, so the `actions/checkout@v4` action is the [`actions`](https://github.com/actions) GitHub organization, the [`checkout`](https://github.com/actions/checkout) repository within that organization, and the [`v4`](https://github.com/actions/checkout/releases/tag/v4) tagged release.

That suggests to me that probably each caching operation would need to be a separate repository inside of the `isofit` GitHub organization. So you might have an `isofit/action-cache-srtmnet@v1`.

This is the state that `action/checkout@v4` points to:

https://github.com/actions/checkout/blob/b4ffde65f46336ab88eb53be808477a3936bae11/action.yml

Looks like the whole workflow is probably driven by `action.yml`, so not toooooo hard to set up, but will take quite a bit of effort.

It is possible that the `@v4` just points to _any_ git revision, in which case you could point it to a branch inside of the `isofit` repository like `isofit/isofit@action-cache-srtmnet`, but then you would hit versioning problems whenever you wanted to updated it, so the branch would need to be named `action-cache-srtmnet-v1`. Ultimately a separate repository for each action is probably the right move.

Reply:
Agreed! It'd be a nice separate PR that we can set up later. Shouldn't be difficult to implement. I wasn't fully sure if a composite action would have to live in a separate repo, but based on your explanation and reading the GitHub webpage more carefully, I also think that's the way to go.